src/library/scala/collection/immutable/Vector.scala

```diff
@@ -215,6 +216,7 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
     s.dirty = dirty
     s.gotoPosWritable(focus, idx, focus ^ idx)  // if dirty commit changes; go to new pos and prepare for writing
     s.display0(idx & 31) = elem.asInstanceOf[AnyRef]
+    ScalaRunTime.releaseFence()
```

Comment:
is this to publish the mutation of `dirty`? Should we also do this in the constructor?